plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1247,32 +1305,55 @@ public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHan
             rowDelta.validateNoConflictingDataFiles();
         }
 
+        if (isUpdate) {
+            rowDelta.validateDeletedFiles();
```

Comment:
Why only for update? maybe add a comment.
or can this be unconditional?

Reply:
I'll add a comment. Deleting a row from two commits concurrently shouldn't cause a validation to fail, but deleting a row and updating it concurrently should fail since the update might un-do the delete